added login with google and facebook and some minor changes

Sundwell opened the pull request.

Sundwell committed:
added login with google and facebook and some minor changes

VitaliiFisenko commented on user/views.py:
```diff
@@ -40,4 +41,11 @@ class UserProfileView(DetailView):
+    form_class = UserUpdateForm
+
+    def get_success_url(self):
+        success_url = self.success_url = reverse_lazy('user:profile', args=[self.kwargs['slug']])
```
Double equal is redundant here. Simply return redirect(reverse_lazy('user:profile', args=[self.kwargs['slug']]))

VitaliiFisenko commented on user/models.py:
```diff
@@ -78,7 +89,7 @@ def __str__(self):
         return self.username
 
     class Meta:
-        unique_together = ('username', 'email')
```
Previous decision was right)

VitaliiFisenko commented on user/models.py:
```diff
@@ -1,8 +1,14 @@
 from django.utils.text import slugify
 
 
+def get_image_path(instance, filename):
```
Add doc string here, describe instance type, filename type and type of return value + short function description

VitaliiFisenko commented on user/forms.py:
```diff
@@ -0,0 +1,20 @@
+
+
+class UserUpdateForm(forms.ModelForm):
+    activities = forms.ModelMultipleChoiceField
```
Here you call widget, pass parameters in ModelMultipleChoiceField constructor

VitaliiFisenko commented on templates/user/update.html:
```diff
@@ -0,0 +1,10 @@
+{% extends 'base.html' %}
+{% load bootstrap4 %}
```
You should to load bootstrap4 once in base template

VitaliiFisenko commented on Food_Delivery_App/settings.py:
```diff
@@ -25,14 +25,15 @@
 DEBUG = True
 
-ALLOWED_HOSTS = []
+# ALLOWED_HOSTS = '*'
```
Delete comments and avoid it in future

Sundwell committed:
added django-select2 to user profile edit page.
Fixed registration with google and facebook when user was getting wrong username and hadn't email.
Added authorization by email too.
And some other minor changes

Sundwell committed:
added '()' to Select2MultipleWidget

Sundwell committed:
fixed some mistakes

Sundwell committed:
minor improves in code

Sundwell committed:
minor improves in code
